aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -177,34 +176,34 @@ pub trait TDelayedFieldView {
     /// 1. The resource is read during the transaction execution.
     /// 2. The resource is not present in write set of the VM Change Set.
     /// 3. The resource has a delayed field in it that is part of delayed field change set.
-    /// We get these resources and include them in the write set of the transaction output.
+    /// We get the keys of these resources and metadata to include them in the write set
+    /// of the transaction output after value exchange.
     fn get_reads_needing_exchange(
         &self,
-        delayed_write_set_keys: &HashSet<Self::Identifier>,
+        delayed_write_set_ids: &HashSet<Self::Identifier>,
         skip: &HashSet<Self::ResourceKey>,
-    ) -> Result<BTreeMap<Self::ResourceKey, (Self::ResourceValue, Arc<MoveTypeLayout>)>, PanicError>;
+    ) -> Result<
+        BTreeMap<Self::ResourceKey, (StateValueMetadata, u64, Arc<MoveTypeLayout>)>,
```

Comment:
Having a comment what the u64 represents would be helpful.

I thought we generally prefer generic types over concrete types. Just curious about the reason to replace ResourceValue with a concrete type.

Reply:
Will add a comment - this is size that was computed by the caller I believe. We might even remove this size at some point (@igor-aptos can confirm)

Reply:
Why do we need to output the resource size here? Is that required somewhere?

Reply:
the callers.